tripy/tripy/backend/compiler_api.py

```diff
@@ -33,6 +34,108 @@
+
+    This class is a wrapper around the underlying stream object, allowing management of CUDA streams.
+
+    Args:
+        priority (int): Assign priority for the new stream. Lower number signifies higher priority.
+
+    .. code-block:: python
+        :linenos:
+        :caption: Create new streams
+
+        streamA = tp.Stream()
+        streamB = tp.Stream()
+        defaultStream = tp.Stream.get_default_stream()
+        assert streamA != streamB
+        assert defaultStream != streamB
+
+    .. code-block:: python
+        :linenos:
+        :caption: Use streams in execution
+
+        linear = tp.Linear(2, 3)
+        compiler = tp.Compiler(linear)
+
+        compiled_linear = compiler.compile(tp.InputInfo((2, 2), dtype=tp.float32))
+
+        a = tp.ones((2, 2), dtype=tp.float32)
+
+        compiled_linear.stream = tp.Stream()
+        out = compiled_linear(a)
+        assert cp.array_equal(cp.from_dlpack(out), cp.from_dlpack(linear(a)))
+    """
+
+    _default_stream = None  # class variable
+
+    _active_cuda_stream = None
+
+    def __init__(self, priority=0):
+        if priority != 0:
+            raise_error(
+                "Incorrect stream priority",
+                [f"Stream priority can only be 0 until #172 is fixed, got priority={priority}."],
+            )
+        from tripy.backend.mlir.utils import MLIRRuntimeClient
+
+        self._active_cuda_stream = MLIRRuntimeClient().create_stream()
+
+    @classmethod
+    def get_default_stream(cls):
+        """Get the default stream, create it if it doesn't exist."""
+        if cls._default_stream is None:
+            cls._default_stream = cls()
+        return cls._default_stream
+
+    @classmethod
+    def synchronize(cls):
+        """Synchronize the default stream, blocking until all operations in this stream are complete."""
+        assert (
+            cls._default_stream != None
+        ), f"Expect default stream to be created before calling tp.Stream.synchronize()"
+        cls._default_stream.synchronize()
+
+    def synchronize(self):
+        """Synchronize the stream, blocking until all operations in this stream are complete."""
+        self._active_cuda_stream.sync()
+
+    def __del__(self):
+        del self._active_cuda_stream
+
+    def __eq__(self, other):
+        if not isinstance(other, Stream):
+            return False
+
+        if not (hasattr(self, "_active_cuda_stream") and hasattr(other, "_active_cuda_stream")):
+            return False
+
+        return self._active_cuda_stream == other._active_cuda_stream
+
+    def __repr__(self):
+        return f"<Stream(id={id(self)}, default={self == Stream._default_stream})>"
+
+    def __hash__(self):
+        return hash(id(self))
+
+    @classmethod
+    def cleanup_default_stream(cls):
+        if cls._default_stream:
+            # TODO: expose methods from mlir-tensorrt python bindings to destroy stream
+            cls._default_stream = None
+        cls._active_stream = None
```

Comment:
I don't think this is needed

Reply:
Do you mean the method is not required or setting the active_stream to None?

Reply:
Method is not required, as this will happen on program exit anyway.

Reply:
We might need this to correctly destroy stream object created in `mlir-tensorrt`.